DEVELOPMENT.md

```diff
@@ -0,0 +1,135 @@
+# Development
+
+**Table of contents:**
+
+1. [Getting started](#getting-started)
+1. [Build the project](#build-the-project_)
+1. [Logging](#logging)
+1. [Error handling](#error-handling)
+
+## Getting started
+
+1. [Ramp up your development environment](#ramp-up)
+1. [Get familiar with Go language](#go-basics)
+1. Create [a GitHub account](https://github.com/join)
+1. Setup [GitHub access via SSH](https://help.github.com/articles/connecting-to-github-with-ssh/)
+1. [Create and checkout a repo fork](#checkout-your-fork)
+1. Optional: [Get Jenkins related environment](#jenkins-environment)
+1. Optional: [Get familiar with Jenkins Pipelines as Code](#jenkins-pipelines)
+
+### Ramp up
+
+First you need to set up an appropriate development environment:
+
+Install Go, see [GO Getting Started](https://golang.org/doc/install)
+
+Install an IDE with Go plugins, see for example [Go in Visual Studio Code](https://code.visualstudio.com/docs/languages/go)
+
+### Go basics
+
+In order to get yourself started, there is a lot of useful information out there.
+
+As a first step to take we highly recommend the [Golang documentation](https://golang.org/doc/) especially, [A Tour of Go](https://tour.golang.org/welcome/1)
+
+We have a strong focus on high quality software and contributions without adequate tests will not be accepted.
+There is an excellent resource which teaches Go using a test-driven approach: [Learn Go with Tests](https://github.com/quii/learn-go-with-tests)
+
+### Checkout your fork
+
+The project uses [Go modules](https://blog.golang.org/using-go-modules). Thus please make sure to **NOT** checkout the project into your [`GOPATH`](https://github.com/golang/go/wiki/SettingGOPATH).
+
+To check out this repository:
+
+1. Create your own
+    [fork of this repo](https://help.github.com/articles/fork-a-repo/)
+1. Clone it to your machine, for example like:
+
+```shell
+mkdir -p ${HOME}/projects/jenkins-library
```

Comment:
Not sure if we should describe the standard github workflow here. Maybe it would be better to provide a link: https://gist.github.com/Chaser324/ce0505fbed06b947d962 (this link here is just an example, maybe someone known an even better resource).

By the way: most of the contributers here does no follow that appraoch, but prefer to work directly in the upstream repo (of course only possible for team members). That comes with some disadvantages (e.g. everyone creating a fork gets all the branches from the upstream in her/his repo). Maybe it would be an good idea to follow the normal github workflow in general, but that is another story ...